luatest/server.lua

```diff
@@ -334,7 +334,13 @@ local function wait_for_condition(cond_desc, server, func, ...)
     local deadline = clock.time() + WAIT_TIMEOUT
     while true do
         if not server.process:is_alive() then
-            error(('Process is terminated when waiting for "%s" condition for server (alias: %s, workdir: %s, pid: %d)')
+            -- The error does not appear at our stack level because something sends
+            -- signum with 14 status code and the external signal handler is called.
```

Comment:
Well, while digging tarantool/tarantool#8748 we didn't receive `SIGALRM` with @Buristan, so this line is little related to the actual case. I'd rather mentioned two facts:
1. The resulting string length/hash matters (at least, its change provides different results)
2. It's likely related to the recent changes in LuaJIT unwinder scope (I suggest to try your case either with Tarantool 3.0 before tarantool/tarantool@cde911d or with the recent 2.11 release).

Otherwise, LGTM.

Reply:
Now it is clear to us that this is not a _Lua interpreter_ problem, but a defect in the process completion logic.

See [comment](https://github.com/tarantool/tarantool/issues/8748#issuecomment-1613320855).